### PR TITLE
FIX: Use bash on macOS, too

### PR DESCRIPTION
**Pull request by larsoner**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] ~~Reset the build number to `0` (if the version changed)~~
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Closes https://github.com/mne-tools/mne-installers/issues/93
Closes https://github.com/mne-tools/mne-installers/issues/101

@hoechenberger any reason you didn't use bash on macOS the same way as Linux? In theory I think this should work. Not totally sure how to test it, though...

I would be great if at least on macOS and linux we could locally build this package and then build an `mne-installers` using our local build of the package.

### CODE REVIEW

**Comment by hoechenberger** [comment]
I don't remember … but we should use `zsh` on macOS as this has been the default shell for many years now

**Comment by hoechenberger** [comment]
… I guess I should probably document how you can test this locally …

File: recipe/menu/menu.json [hunk 1]
```diff
@@ -69,7 +69,7 @@
         "command": [
           "%SystemRoot%\\system32\\cmd.exe",
           "/K",
-          "{{ SCRIPTS_DIR }}\\activate.bat"
+          "{{ MENU_DIR }}\\mne_open_prompt.bat"
```
[the remark this answers is not in the document]
Comment (hoechenberger) [reply recipe/menu/menu.json:72]: ok looks good then! let's not touch this then for now

File: recipe/menu/open_prompt.sh [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
+# This is used to initialize the bash prompt on macOS and Linux.
 
-. ~/.bashrc
-. __PREFIX__/bin/activate
-echo "Using Python: `which python`"
-echo "This is MNE-Python `mne --version`"
+if [[ -f ~/.bashrc ]] && [[ ${OSTYPE} != 'darwin'* ]]; then
+    source ~/.bashrc
+fi
+source __PREFIX__/bin/activate
```
[the remark this answers is not in the document]
Comment (larsoner) [reply recipe/menu/open_prompt.sh:8]: I think in theory it should not be necessary now if it wasn't before, so let's try this separately later

File: recipe/menu/open_prompt.bat [hunk 3]
```diff
@@ -0,0 +1,10 @@
+:: This is used to initialize the bash prompt on Windows.
+@ECHO OFF
+
+call %SCRIPTS_DIR%\Activate.bat
```
Comment (larsoner) [inline recipe/menu/open_prompt.bat:4]: Maybe this should be 
```suggestion
call %__PREFIX__%\Scripts\Activate.bat
```
WDYT @hoechenberger ?

Comment (larsoner) [reply recipe/menu/open_prompt.bat:4]: ... based on how `open_prompt.sh` works, I think it should be. I'll commit but feel free to correct me if I'm wrong

**Pull request merged**